File: Kap13/C#/Listing62_64/mygraphs/Kreis.cs
namespace mygraphs {
  public class Kreis : ZweiDElm {
     protected double radius;
     
    public Kreis(int initRandfarbe, int initFuellfarbe, double radius) : base(initRandfarbe, initFuellfarbe) {
      this.radius = radius;
      berechneUmfang();
      berechneFlaeche();
      Console.WriteLine("Kreis erzeugt.");
    }
    
    protected override void berechneUmfang() {
      umfang = 2*radius * Math.PI;
    }

    protected override void berechneFlaeche() {
      umfang = radius * radius * Math.PI;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }
    
    public double getRadius() {
      return radius;
    }

      public override void printInfo() {
      base.printInfo();
      Console.WriteLine("Radius: " + radius);
    }
  }
}

File: Kap13/Java/Listing38/mygraphs/Kreis.java
package Kap13.Java.Listing38.mygraphs;

public class Kreis extends ZweiDElm {
    protected double radius;
     
    public Kreis(int initRandfarbe, int initFuellfarbe, double radius) {
      super(initRandfarbe, initFuellfarbe);
      this.radius = radius;
      berechneUmfang();
      berechneFlaeche();
      System.out.println("Kreis erzeugt.");
    }

    protected void berechneUmfang() {
      umfang = 2*radius * Math.PI;
    }

    protected void berechneFlaeche() {
      flaeche = radius * radius * Math.PI;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }
    
    public double getRadius() {
      return radius;
    }

      public void printInfo() {
      super.printInfo();
      System.out.println("Radius: " + radius);
    }
  }	
  